src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -174,78 +185,55 @@ def process_files(self) -> Iterable[pyarrow.table]:
+                if dt_part not in self.data_parts:
+                    self.data_parts[dt_part] = TableData()
+
+                self.data_parts[dt_part].files.append(result_filename)
+
+                if self.data_parts[dt_part].table is None:
+                    self.data_parts[dt_part].table = (
+                        self.detail.transform_for_write(rt_data)
+                    )
+                else:
+                    self.data_parts[dt_part].table = pyarrow.concat_tables(
+                        [
+                            self.data_parts[dt_part].table,
+                            self.detail.transform_for_write(rt_data),
+                        ]
+                    )
+
+                yield from self.yield_check(process_logger)
 
         # yield any remaining tables
-        yield from self.yield_check(-1, process_logger)
+        yield from self.yield_check(process_logger, min_rows=0)
 
         process_logger.add_metadata(file_count=0, number_of_rows=0)
         process_logger.log_complete()
 
     def yield_check(
-        self, yield_threshold: int, process_logger: ProcessLogger
+        self, process_logger: ProcessLogger, min_rows: int = 1_000_000
     ) -> Iterable[pyarrow.table]:
         """
-        yield all tables in the table_groups map that have been sufficiently
+        yield all tables in the data_parts map that have been sufficiently
         processed.
 
-        gtfs realtime files are processed chronologically in a thread pool and
-        table groups are collections of gtfs realtime tables that are from the
-        same hour. if they were being processed in series we could yield a
-        table as soon as a realtime file from the next hour was processed.
-        instead, ensure that a sufficient number for files from the next hour
-        have been processed.
-
-        @yield_threshold - how many files from the next hour have to be
-            processed before considering _this_ hour complete.
+        @min_rows - how many rows the table must have to be yielded
         @process_logger - a process logger for the conversion process. log a
             completion and reset before a file is yielded.
 
-        @yield pyarrow.table - a concatenated table of all the gtfs realtime
-            data over the corse of an hour.
+        @yield pyarrow.table - a concatenated table of gtfs realtime data.
         """
-        for iter_ts in list(self.table_groups.keys()):
-            if self.table_groups[iter_ts].next_hr_cnt > yield_threshold:
-                self.archive_files += self.table_groups[iter_ts].files
-
-                table = self.table_groups[iter_ts].table
-
-                assert table is not None
+        for iter_ts in list(self.data_parts.keys()):
+            table = self.data_parts[iter_ts].table
+            if table is not None and table.num_rows > min_rows:
+                self.archive_files += self.data_parts[iter_ts].files
```

Comment:
no more `next_hr_count` yield table if not None and over `min_rows` 1_000_000 by default, reduced to 0 when all files are processed.